### ISSUE TITLE
Invite/Trade/Message buttons stopped working

### ISSUE THREAD

**Issue by garralab** [Exslims/MercuryTrade#90]
Hi, first of all thanks! Great tool! I've been without for a few day and I already feel lost... I'm at the point that I don't even feel like to play until I solved this...

MT was working perfectly. All of a sudden all buttons stopped working. Game was running, so I don't think it's an update or the way it is runned. Anyway I tryed the exe launcher runned as Admin, nothing changed.
So I tried an hard reset. Now the "Still interested..." button works, but not all the times (so maybe it was working even before). On average it sends the message every 3 attempts. It's almost like it has problems getting the focus of the application. Any other button does nothing, no matter how many time I try. All except the send message one that at least opens the chat, but doesn't paste the character name.

I'm on win7.

Thank you, and kutvgw!!

**Comment by Exslims** [comment]
Hi, could you provide app.log file from %userprofile%/appdata/local/MercuryTrade folder?

**Comment by Exslims** [comment]
it looks like issue with windows creators update, can you try to update your videocard drivers? may be you have some utilities like MSI Autoburner

**Comment by garralab** [comment]
Hi, creator shouldn't be an issue since I'm on win 7.
Attaching logs, but nothing there at the times it didn't work. 
I'll try to update my video driver next.
[app.txt](https://github.com/Exslims/MercuryTrade/files/1081486/app.txt)

**Comment by garralab** [comment]
No new drivers

**Comment by garralab** [comment]
Narrowed it down finally. It's a conflict with PoE-TradeMacro since version 2.2.0. I always used both programs together so I tried load previous versions and with POE-TM 2.1.1 everything works, from 2.2.0 onward every chat command from MercuryTrade is blocked. I don't really know how this is possible. I'm gonna post a bug report to them. Really hope I won't to choose between the two. It would be a really hard choice.

**Comment by Exslims** [comment]
@Eruyome hi, can you see what might conflict?

**Comment by garralab** [comment]
I really don't. I'll take a look at 2.2.0 diff tonight, but I really don't know what can be the problem. By the way, how does Mercury send chat commands? It just send keyboard input (ENTER + Msg + ENTER), right?
I really don't see how this can be prevented...

**Comment by Exslims** [comment]
Very long time ago i removed code which set poe window as foreground window coz there was no sense in this code, and so worked. Maybe if I get it back it will help.

**Comment by garralab** [comment]
Might make sense. Expecially considering the fact that 1% of the time it still works... and win7 might be a bit erratic at times managing foregrounds.

**Comment by Exslims** [comment]
will do so, during the release 3.0 will be a big update and I will add there this code, maybe eruyome also will find something

**Comment by Eruyome** [comment]
I can take a look. Are you still guessing that this removed code would fix that or is it confirmed?

**Comment by Eruyome** [comment]
Here's the diff of 2.1.1 and 2.2.0, https://github.com/poe-trademacro/poe-trademacro/compare/v2.1.1%E2%80%A6v2.2.0.
I can't see any change that I would associate whith this issue. The only change I could remotely think of would be the one to the function `HighlightItems()` in `resources/ahk/POE-ItemInfo.ahk` since it involves the clipboard and copy pasting to the search fields. But none of this would be triggered unless the ctrl + f or ctrl + f + alt hotkeys are triggered.

@garralab What TradeMacro versions have you tested since 2.1.1?

**Comment by garralab** [comment]
2.2.0 and 2.3.4
then 2.3.3, 2.3.2 and 2.3.1 to see if one had no problem, but did so a bit in a hurry. I can check again if you think one should be different, or just want to be sure.

**Comment by Eruyome** [comment]
I have no reason to believe that one would be different since I don't see a reason for that issue in the first place. Making sure could help though considering that I have no idea.

**Comment by garralab** [comment]
Well I tried something different. I got the git repo and placed myself on 2.2.0 tag, and was going to try and revert code to 2.1.1 a hunk at a time. Luckily I tested 2.2.0 before starting, because it worked!!! So I put myself on master head and it works... tryed the distributed versions again and now they all work too. Obviously I also didn't change a thing in mercury...

At least now I can use latest version of both, but I really don't like this... it was systematich and reproduceable, it stayed with me for a couple of weeks, through cleans, restarts, reinstall, poe patches, mercury updates and poe trade updates... I would have really prefered to unerstand what it was.

If it comes back I'll go back to revert one hunk at a time. Until then forget about this, you got much better things to do. Never been so furious for not having a bug.
